lib/fizzy/execute.cpp

```diff
@@ -109,6 +109,51 @@ inline bool store_into_memory(
     return true;
 }
 
+// Check that exception is one of the types expected to be thrown from bytes::resize().
+// We catch ... in memory.grow implementation for the sake of smaller binary code and assert it's
+// one of expected exceptions.
+[[maybe_unused]] bool is_resize_exception(std::exception_ptr exception) noexcept
+{
+    try
+    {
+        std::rethrow_exception(exception);
+    }
+    catch (const std::bad_alloc&)
+    {
+        return true;
+    }
+    catch (const std::length_error&)
+    {
+        return true;
+    }
+    catch (...)
+    {
+        return false;
+    }
+}
+
+inline uint32_t grow_memory(
```

Comment:
If we are going with this helper, please document it. What is important to document that it returns page count prior to expansion.